sklearn/preprocessing/function_transformer.py

```diff
@@ -0,0 +1,52 @@
+from ..base import BaseEstimator, TransformerMixin
+from ..utils import check_array
+
+
+def _identity(X):
+    """The identity function.
+    """
+    return X
+
+
+class FunctionTransformer(BaseEstimator, TransformerMixin):
+    """Constructs a transformer from an arbitrary callable.
+
+    Note: If a lambda is used as the function, then the resulting
+    transformer will not be pickleable.
+
+    Parameters
+    ----------
+    func : callable, optional default=None
+        The callable to use for the transformation. This will be passed
+        the same arguments as transform, with args and kwargs forwarded.
+        If func is None, then func will be the identity function.
+
+    validate : bool, optional default=True
+        Indicate that the input X array should be checked before calling
+        func. If validate is false, there will be no input validation.
```

Comment:
Note that this will ensure the input is a non-empty, 2-dimensional array (or sparse matrix) of finite numbers.